### DISCUSSION TITLE
Exposing validateJWT function

### DISCUSSION THREAD

**Discussion by duckboy81** [panva/openid-client#580] [Ideas]
Hey there, thanks for an amazing library!

I was wondering if there was a possibility for you to expose the `validateJWT` function in the `BaseClient`. It would seem this function is perfect to use since it is already setup to obtain updated public keys published by the OP (among other things).

I see the discussion of offline validation here #568, but I wonder if it'd be easier to use the existing infrastructure in this library instead of building my own code using `jose`.

**Comment by panva** [comment]
`validateJWT` is not and will not be a part of the public documented API, you're more likely to footgun yourself with this internal implementation than with a simple API like `jose`'s

```js
import * as jose from 'jose';

const JWKS = jose.createRemoteJWKSet(new URL(issuer.metadata.jwks_uri))

const { payload, protectedHeader } = await jose.jwtVerify(jwt, JWKS, {
  typ: 'jwt+at',
  issuer: 'urn:example:issuer',
  audience: 'urn:example:audience',
})

// assert payload claims you require such as scope
```

**Reply by duckboy81** [reply]
Easy enough. Thanks for your time.
